vendor/src/github.com/docker/engine-api/types/container/config.go

```diff
@@ -5,6 +5,27 @@ import (
 	"github.com/docker/go-connections/nat"
 )
 
+// HealthConfig holds configuration settings for the HEALTHCHECK feature.
+type HealthConfig struct {
+	// Test to perform to check container is healthy.
+	// An empty slice means to inherit the default.
+	// Otherwise, Cmd[0] is "NONE" or "CMD".
+	Test strslice.StrSlice `json:",omitempty"`
+
+	// Nil means to inherit.
+	GracePeriod *float64 `json:",omitempty"` // Time after container start to assume failure means still starting.
+	Interval    *float64 `json:",omitempty"` // Time to wait between checks.
+	Timeout     *float64 `json:",omitempty"` // Time to wait before considering the check to have hung.
+
+	// Number of consecutive failures needed to consider a container as unhealthy.
+	// Zero means inherit.
+	Retries uint `json:",omitempty"`
+
+	// Stop container on entering unhealthy state.
+	// Nil means inherit.
+	ExitOnUnhealthy *bool `json:",omitempty"`
```

Comment:
Using a boolean pointer in Go is an anti-pattern.

There are only two options. Exit on unhealthy or not. The default is to not (bools are initilized to false by default). Use ordering to achieve this behavior:

1. Grab image health check. If image specifies to exit, set to true.
2. Apply flags. Set based on flag.

None of this requires a pointer. All you need to know is whether or not the flag was set. If it was, apply it. Adding a pointer here just makes a mess.

Either way, this belongs on restart policy, rather than the health check.